cmd/spiffe-aws-assume-role/cli/cli.go

```diff
@@ -40,24 +53,41 @@ func (c *CredentialsCmd) Run() error {
 	return err
 }
 
-var CLI struct {
+type CLI struct {
 	Credentials CredentialsCmd `kong:"cmd,help:'print credentials in a format usable as an AWS credentials_process'"`
 }
 
-func Run(args []string) error {
-	p, err := kong.New(&CLI)
-	if err != nil {
-		return err
+func RunWithDefaultContext(args []string) error {
+	context := &CliContext{
+		JWTSourceProvider: credentials.StandardJWTSourceProvider,
+		STSProvider:       credentials.StandardSTSProvider,
 	}
 
-	ctx, err := p.Parse(args)
+	return Run(context, args)
+}
+
+func Run(context *CliContext, args []string) error {
+	ctx, err := parse(args)
 	if err != nil {
 		return err
 	}
 
-	if err = ctx.Run(); err != nil {
+	if err = ctx.Run(context); err != nil {
 		return err
 	}
 
 	return nil
 }
+
+func newKong(cli *CLI) (*kong.Kong, error) {
+	return kong.New(cli, kong.NamedMapper(util.Iso8601DurationMapperType, util.Iso8601DurationMapper{}))
+}
+
+func parse(args []string) (*kong.Context, error) {
+	parser, err := newKong(&CLI{})
```

Comment:
I moved the Kong instantiation logic to a factory method to aid in testing (we need direct access to this object in order to disable the exit call after help is displayed).